### PR TITLE
Closes #1177 N+1 queries when loading live actions

### LINKED ISSUE #1177: N+1 queries when loading live actions
**Describe the bug**

The "Live Actions" view loads really slow for me (localhost, million+ events in DB).

It seems that it makes a new query for every event:

screencast 2020-07-09 13-50-05

This could be optimized somehow :)

### PR DESCRIPTION
**Pull request by 14MR**
## Changes

- filter in-memory, instead of generation n+1 queries 

## Checklist
- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

**Comment by 14MR** [comment]
Looks like some infrastructure problem. 
image

File: posthog/api/event.py [hunk 1]
```diff
@@ -212,9 +212,11 @@ def actions(self, request: request.Request) -> response.Response:
             if event.pk in ids_seen:
                 continue
             ids_seen.append(event.pk)
-            for action in event.action_set.filter(deleted=False):
-                event.action = action
-                matches.append(event)
+            for action in event.action_set.all():
+                # do not use .filter(), because it causes N+1 query problem. Filter just in memory
```
Comment (timgl) [inline posthog/api/event.py:216]: Thanks for this! An easier solution might be to do this filtering in Prefetch on line 207:
` .prefetch_related(Prefetch("action_set", queryset=Action.objects.filter(deleted=False).order_by("id")))[0:101]`

I think that should work (thought could be wrong!)

Comment (14MR) [reply posthog/api/event.py]: Thanks for the tip, tried it, looks like it works correctly

File: posthog/api/event.py [hunk 2]
```diff
@@ -204,15 +204,15 @@ def actions(self, request: request.Request) -> response.Response:
...
+            .prefetch_related(Prefetch("action_set", queryset=Action.objects.filter(deleted=False).order_by("id")))[0:101]
         )
         matches = []
         ids_seen: List[int] = []
         for event in events:
             if event.pk in ids_seen:
                 continue
-            ids_seen.append(event.pk)
-            for action in event.action_set.filter(deleted=False):
+            ids_seen.append(event.pkm)
```
Comment (timgl) [inline posthog/api/event.py:214]: This should be `.pk`

**Comment by timgl** [comment]
@rudasn The reason is when using filter we were doing a separate query for each loop. Now we pre-filter the queryset

**Comment by rudasn** [comment]
> @rudasn The reason is when using filter we were doing a separate query for each loop. Now we pre-filter the queryset

Got it, thanks!

**Pull request merged**